src/smvlang/parser.y

```diff
@@ -130,6 +130,41 @@ static void j_binary(YYSTYPE & dest, YYSTYPE & op1, const irep_idt &id, YYSTYPE
 
 /*******************************************************************\
 
+Function: merge_complex_identifier
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+irep_idt merge_complex_identifier(const exprt &expr)
+{
+  if(expr.id() == ID_symbol)
+    return to_symbol_expr(expr).get_identifier();
+  else if(expr.id() == ID_member)
+  {
+    auto &member_expr = to_member_expr(expr);
+    return id2string(merge_complex_identifier(member_expr.compound())) + '.' + id2string(member_expr.get_component_name());
```

Comment:
In CBMC's field sensitivity encoding we chose `..` as separator to avoid confusing humans as to whether they see an identifier or an expression. Would that also be a possible concern (and solution) here?

Reply:
I did worry about this, but SMV doesn't have a `struct` type.